### DISCUSSION TITLE
Override request validation exceptions documentation example not outputting as expected

### DISCUSSION THREAD

**Discussion by seantalon** [fastapi/fastapi#12125] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, HTTPException
from fastapi.exceptions import RequestValidationError
from fastapi.responses import PlainTextResponse
from starlette.exceptions import HTTPException as StarletteHTTPException

app = FastAPI()


@app.exception_handler(StarletteHTTPException)
async def http_exception_handler(request, exc):
    return PlainTextResponse(str(exc.detail), status_code=exc.status_code)


@app.exception_handler(RequestValidationError)
async def validation_exception_handler(request, exc):
    return PlainTextResponse(str(exc), status_code=400)


@app.get("/items/{item_id}")
async def read_item(item_id: int):
    if item_id == 3:
        raise HTTPException(status_code=418, detail="Nope! I don't like 3.")
    return {"item_id": item_id}
```

### Description

Run the code example in this documentation section: https://fastapi.tiangolo.com/tutorial/handling-errors/#override-request-validation-exceptions

The expected output according to the docs is:
1 validation error
path -> item_id
  value is not a valid integer (type=type_error.integer)
  
  Instead, we get the json as in the normal response:
  [{'type': 'int_parsing', 'loc': ('path', 'item_id'), 'msg': 'Input should be a valid integer, unable to parse string as an integer', 'input': 'foo'}]
  
  If I str(e) a ValidationError in a repl, I do see the proper format.  It may be a problem with RequestValidationError __repr__?  Not sure, I'm not an expert by any means.

### Operating System

Linux, Windows

### Operating System Details

Tried Windows and Ubuntu 22.04, 24.04

Tried python 3.12 initially, 3.11, 3.10

### FastAPI Version

0.112.2

### Pydantic Version

2.8.2

### Python Version

3.12

### Additional Context

![image](https://github.com/user-attachments/assets/4e671285-9aac-42f0-9494-a012c59df674)

**Comment by ecly** [comment, marked as answer]
@itsavaren I believe this is a remnant of `RequestValidationError` no longer being an actual subclass of `ValidationError` since Pydantic 2.0. It is being discussed and fixed in: https://github.com/fastapi/fastapi/pull/11542

I have highlighted the need to update this bit of the documentation on the PR here: https://github.com/fastapi/fastapi/pull/11542/files/b91428de008aa0c79729006068021b84d1575feb#r1744156615

**Reply by seantalon** [reply]
Thanks for the insight!

**Reply by tamird** [reply]
The problem is indeed a missing implementation of `__str__`. See https://github.com/fastapi/fastapi/pull/12972.
